src/app/server/Mdns.cpp

```diff
@@ -110,9 +110,7 @@ CHIP_ERROR AdvertiseOperational()
 
     auto & mdnsAdvertiser = chip::Mdns::ServiceAdvertiser::Instance();
 
-    ReturnErrorOnFailure(mdnsAdvertiser.Advertise(advertiseParameters));
-
-    return mdnsAdvertiser.Start(&chip::DeviceLayer::InetLayer, chip::Mdns::kMdnsPort);
+    return mdnsAdvertiser.Advertise(advertiseParameters);
```

Comment:
Would the missing start still have things work when using minmdns?

I wonder if here we are trading ability to start platform mdns with ability to start minmdns.

Reply:
Line 529 in `src/app/server/Server.cpp` says:

```
  527 // Starting mDNS server only for Thread devices due to problem reported in issue #5076.
  528 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
  529     app::Mdns::StartServer();
  530 
```
And this call is in `InitServer` so the `ServiceAdvertiser::Start()` method will be always called before advertising.

Reply:
In fact we found reoving this line also a must for minmdns to function on ESP32. Repeatedly closing and starting the server will drain the udp endpoint pool since `Release` will be defered to the next event loop.

Reply:
Ah, that may explain out of udp endpoints reported in #5759 .